ui/src/app/pages/login/sign-in.tsx

```diff
@@ -71,99 +60,188 @@ const styles = {
+          return <AccountCreationTos
+            pdfPath='/assets/documents/terms-of-service.pdf'
+            onComplete={() => this.setState({
+              termsOfServiceVersion: 1,
+              currentStep: SignInStep.ACCOUNT_CREATION
+            })}/>;
+        case SignInStep.ACCOUNT_CREATION:
+          return <AccountCreation invitationKey={this.state.invitationKey}
+                                  profile={this.state.profile}
+                                  onComplete={(profile: Profile) => this.setState({
+                                    profile: profile,
+                                    // Skip over the demographic survey if new-style form isn't enabled.
+                                    currentStep: enableNewAccountCreation ? SignInStep.DEMOGRAPHIC_SURVEY :
+                                      SignInStep.SUCCESS_PAGE
+                                  })}/>;
+        case SignInStep.DEMOGRAPHIC_SURVEY:
+          return <AccountCreationSurvey
+            profile={this.state.profile}
+            invitationKey={this.state.invitationKey}
+            termsOfServiceVersion={this.state.termsOfServiceVersion}
+            onComplete={(profile: Profile) => this.setState({
+              profile: profile,
+              currentStep: SignInStep.SUCCESS_PAGE
+            })}
+            onPreviousClick={(profile: Profile) => this.setState({
+              profile: profile,
+              currentStep: SignInStep.ACCOUNT_CREATION
+            })}/>;
+        case SignInStep.SUCCESS_PAGE:
           return <AccountCreationSuccess profile={this.state.profile}/>;
         default:
           return;
       }
     }
 
-    setCurrentStep(nextStep: Step) {
-      this.setState({
-        currentStep: nextStep
-      });
-    }
-
-    onKeyVerified(invitationKey: string) {
-      this.setState({
-        invitationKey: invitationKey,
-        currentStep: {stepName: 'accountCreation'}
-      });
-    }
-
-    setProfile(profile, currentStep) {
-      this.setState({
-        profile: profile,
-        currentStep: currentStep
-      });
-    }
-
-
-
     render() {
-      const {stepName, backgroundImages} = this.state.currentStep;
-      const maxWidth = backgroundImages === undefined ? '100%' : '41.66667%';
-      return <div style={styles.signedInContainer}>
-        <FlexColumn style={{width: '100%'}}>
-          <div data-test-id='template'
-               style={styles.template(this.props.windowSize, backgroundImages)}>
-            <img style={{height: '1.75rem', marginLeft: '1rem', marginTop: '1rem'}}
-                 src={headerImg}/>
-            <div style={{flex: `0 0 ${maxWidth}`,
-              maxWidth: maxWidth, minWidth: '25rem'}}>
-              {this.nextDirective(stepName)}
-            </div>
-          </div>
+      const backgroundImages = StepToImageConfig[this.state.currentStep];
+      return <FlexColumn style={styles.signInContainer} data-test-id='sign-in-container'>
```

Comment:
Dev note: it was quite a challenge to get the sign-in and Terms of Service components to be styled correctly to allow a "flexible" PDF panel with a sticky footer at the bottom to contain the checkboxes & next button. Ultimately, I took an approach of using nested FlexColumn components all the way down... this seemed to be minimally disruptive to the existing styling, though I haven't extensively tested this on very differently-sized screens. A bit of tire-kicking against my staged App Engine version would be very helpful here.

Reply:
I couldn't break it - looks pretty nice to me.